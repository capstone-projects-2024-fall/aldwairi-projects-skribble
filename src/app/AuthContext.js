import React, { createContext, useState } from "react";

/**
 * AuthContext is a context that provides authentication-related state
 * (such as the session token) and functions (like setting the session token)
 * to the components within the application.
 *
 * @type {React.Context<{ sessionToken: string | null, setSessionToken: React.Dispatch<React.SetStateAction<string | null>> }>}
 */
export const AuthContext = createContext();

/**
 * AuthProvider component is a wrapper that provides the authentication context
 * to the children components. It manages the session token state, allowing
 * components to access and update it.
 * 
 * - Provides the `sessionToken` and `setSessionToken` via context.
 * - Any component wrapped in this provider can access or modify the authentication state.
 * 
 * @param {{ children: React.ReactNode }} param0 - The child components to be wrapped.
 * @param {React.ReactNode} param0.children - The child elements that will have access to the authentication context.
 * @returns {JSX.Element} The rendered AuthProvider component with the context provider.
 */
export const AuthProvider = ({ children }) => {
  const [sessionToken, setSessionToken] = useState(null);

  return (
    <AuthContext.Provider value={{ sessionToken, setSessionToken }}>
      {children}
    </AuthContext.Provider>
  );
};
