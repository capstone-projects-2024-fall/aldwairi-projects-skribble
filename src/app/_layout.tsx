import React from "react";
import { Stack } from "expo-router";

export default function RootLayout() {
  return (
    <Stack>
      <Stack.Screen name="index" options={{ title: "LogIn" }} />
      <Stack.Screen name="home" options={{ title: "Home" }} />
    </Stack>
  );
}
