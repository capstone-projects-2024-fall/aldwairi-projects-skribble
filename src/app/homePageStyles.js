import { StyleSheet } from 'react-native';

const styles = StyleSheet.create({
  homeContainer: {
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
  },
  logoContainer: {
      position: 'absolute',
      alignItems: 'center',
  },
  logoContainerWeb: {
      top: 40,
  },
  logoContainerMobile: {
      top: 60,
  },
  logo: {
      resizeMode: 'contain',
  },
  avatar: {
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonContainer: {
      flexDirection: 'row',
      flexWrap: 'wrap',
      justifyContent: 'center',
      marginTop: 20,
  },
  buttonStyle: {
      padding: 18,
      margin: 18,
      borderRadius: 18,
      flexDirection: 'row',
      alignItems: 'center',
      justifyContent: 'center',
      borderWidth: 3,
  },
  buttonText: {
      color: 'black',
      fontSize: 20,
      marginRight: 14,
      fontWeight: 'bold',
  },
  icon: {
      width: 32,
      height: 32,
      marginLeft: 10,
  },
});

export default styles;