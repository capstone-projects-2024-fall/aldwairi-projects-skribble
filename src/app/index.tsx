import React, { useState } from "react";
import { Text, View, TextInput, TouchableOpacity, Alert } from "react-native";
import neo4j from "neo4j-driver";
import { useRouter } from "expo-router";
import styles from "./indexStyles";
import createNeo4jDriver from './utils/databaseSetUp';

export default function LogIn() {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [birthday, setBirthday] = useState("");
  const [parentEmail, setParentEmail] = useState("");
  const [isSignUp, setIsSignUp] = useState(true);
  const [error, setError] = useState(""); // State for error message
  const router = useRouter();

  // Set up the Neo4j driver
  const driver = createNeo4jDriver();

  const handleAuthToggle = () => {
    setIsSignUp(!isSignUp);
    setError(""); // Clear error when switching between Sign Up and Sign In
  };

  const handleAuth = async () => {
    const session = driver.session();
  
    if (isSignUp) {
      if (!email || !password || !birthday) {
        setError("Email, password, and birthday cannot be blank.");
        return;
      }

      const birthDate = new Date(birthday);
      const age = new Date().getFullYear() - birthDate.getFullYear();
      const needsParentalControls = age < 13;

      if (needsParentalControls && !parentEmail) {
        setError("Parent's email is required for users under 13.");
        return;
      }
  
      try {
        // Create a new user or ensure the user exists
        const result = await session.run(
          `MERGE (u:User {email: $email})
           ON CREATE SET 
             u.password = $password,
             u.name = $name,
             u.streak = 0,
             u.coins = 0,
             u.exp = 0,
             u.backgroundColor = '#FFFFFF',
             u.birthday = $birthday,
             u.parentEmail = $parentEmail,
             u.needsParentalControls = $needsParentalControls
           RETURN u`,
          { email, password, name: "New User", birthday, parentEmail, needsParentalControls }
        );
  
        if (result.summary.counters.containsUpdates()) {
          console.log("User signed up successfully.");
          router.push("/homePage");
        } else {
          setError("User already exists.");
        }
      } catch (error) {
        console.error("Failed to sign up user", error);
        setError("An error occurred while signing up. Please try again.");
      } finally {
        await session.close();
        await driver.close();
      }
    } else {
      try {
        // Validate existing user credentials
        const result = await session.run(
          `MATCH (u:User {email: $email, password: $password})
           RETURN u`,
          { email, password }
        );
  
        if (result.records.length > 0) {
          console.log("User signed in successfully.");
          router.push("/homePage");
        } else {
          setError("Invalid email or password.");
        }
      } catch (error) {
        console.error("Failed to sign in user", error);
        setError("An error occurred while signing in. Please try again.");
      } finally {
        await session.close();
        await driver.close();
      }
    }
  };

  return (
    <View style={styles.container}>
      <Text style={styles.title}>{isSignUp ? "Sign Up" : "Sign In"}</Text>

      <TextInput
        style={styles.input}
        placeholder="Email"
        placeholderTextColor="#888888"
        keyboardType="email-address"
        value={email}
        onChangeText={(text) => {
          setEmail(text);
          setError(""); // Clear error message when user starts typing
        }}
      />
      <TextInput
        style={styles.input}
        placeholder="Password"
        placeholderTextColor="#888888"
        secureTextEntry
        value={password}
        onChangeText={(text) => {
          setPassword(text);
          setError(""); // Clear error message when user starts typing
        }}
      />
      {isSignUp && (
        <>
          <TextInput
            style={styles.input}
            placeholder="Birthday (YYYY-MM-DD)"
            placeholderTextColor="#888888"
            value={birthday}
            onChangeText={(text) => {
              setBirthday(text);
              setError(""); // Clear error message when user starts typing
            }}
          />
          <TextInput
            style={styles.input}
            placeholder="Parent's Email (if under 13)"
            placeholderTextColor="#888888"
            keyboardType="email-address"
            value={parentEmail}
            onChangeText={(text) => {
              setParentEmail(text);
              setError(""); // Clear error message when user starts typing
            }}
          />
        </>
      )}

      {/* Display error message */}
      {error ? <Text style={styles.errorText}>{error}</Text> : null}

      <TouchableOpacity onPress={handleAuth} style={styles.button}>
        <Text style={styles.buttonText}>{isSignUp ? "Sign Up" : "Sign In"}</Text>
      </TouchableOpacity>

      <TouchableOpacity onPress={handleAuthToggle}>
        <Text style={styles.toggleText}>
          {isSignUp ? "Already have an account? Sign In" : "New user? Sign Up"}
        </Text>
      </TouchableOpacity>
    </View>
  );
}