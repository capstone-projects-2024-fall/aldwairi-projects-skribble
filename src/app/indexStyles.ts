import { StyleSheet } from "react-native";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: "center",
    alignItems: "center",
    padding: 20,
  },
  title: {
    fontSize: 24,
    marginBottom: 20,
  },
  input: {
    width: "100%",
    height: 50,
    borderColor: "#ddd",
    borderWidth: 1,
    paddingHorizontal: 10,
    borderRadius: 5,
    marginBottom: 15,
  },
  button: {
    backgroundColor: "#007bff",
    padding: 15,
    borderRadius: 5,
    width: "100%",
    alignItems: "center",
  },
  buttonText: {
    color: "#fff",
    fontSize: 16,
  },
  toggleText: {
    color: "#007bff",
    marginTop: 15,
    textAlign: "center",
  },
  errorText: {
    color: "red",
    fontSize: 14,
    marginTop: 10,
    textAlign: "center",
  },
});

export default styles;
