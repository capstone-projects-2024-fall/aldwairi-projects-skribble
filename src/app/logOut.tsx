import AsyncStorage from '@react-native-async-storage/async-storage';
import { router, useRouter } from 'expo-router';

export const logOut = async () => {
    try {
        // Remove the current user data from AsyncStorage
        await AsyncStorage.removeItem('currentUser');
        console.log('User logged out successfully.');

        // Navigate back to the sign-in page
        router.push('/');
    } catch (error) {
        console.error('Failed to log out:', error);
    }
};
