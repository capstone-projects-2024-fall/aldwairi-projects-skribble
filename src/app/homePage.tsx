import React, { useEffect, useState, useContext } from 'react';
import { useRouter } from 'expo-router';
import { View, Text, Image, TouchableOpacity, StyleSheet, Platform, Dimensions, Alert } from 'react-native';
import Svg, { Path } from 'react-native-svg';
import styles from './homePageStyles';
import { avatar_list } from '../assets/avatars/avatarAssets';
import createNeo4jDriver from './utils/databaseSetUp';
import { getDarkerShade } from './utils/colorUtils';
import { AuthContext } from "./AuthContext";

const HomePage: React.FC = () => {
  const router = useRouter();
  const [backgroundColor, setBackgroundColor] = useState<string>('#FFFFFF');
  const [avatarImage, setAvatarImage] = useState(avatar_list[0].avatar_image);
  const [allowAddViewFriends, setAllowAddViewFriends] = useState(true);
  const [enableChat, setEnableChat] = useState(true);
  const { sessionToken } = useContext(AuthContext);
  const screenWidth = Dimensions.get('window').width;

  // Set up the Neo4j driver
  const driver = createNeo4jDriver();

  // load user data
  useEffect(() => {
    const loadUserData = async () => {
      const session = driver.session();
      try {
        const result = await session.run(
          `MATCH (u:User {sessionToken: $sessionToken})
           RETURN 
            u.backgroundColor AS backgroundColor, 
            u.avatarImage AS avatarImage, 
            u.enableChat AS enableChat,
            u.allowAddViewFriends AS allowAddViewFriends`,
          { sessionToken }
        );
        if (result.records.length > 0) {
          // Directly accessing the fields in the result
          const record = result.records[0];
          // Extract values from the result, handling the INTEGER type if necessary
          const backgroundColor = record.get("backgroundColor");
          const avatarImage = record.get("avatarImage");
          const enableChat = record.get("enableChat");
          const allowAddViewFriends = record.get("allowAddViewFriends");

          console.log("User properties:", { backgroundColor, avatarImage, enableChat, allowAddViewFriends }); // Debugging

          setBackgroundColor(backgroundColor || "#FFFFFF");
          setAvatarImage(avatarImage || avatar_list[0].avatar_image);
          setAllowAddViewFriends(allowAddViewFriends);
          setEnableChat(enableChat);
        } else {
          Alert.alert("Error", "User not found.");
        }
      } catch (error) {
        console.error("Failed to load user data", error);
        Alert.alert("Error", "Could not fetch user data.");
      } finally {
        await session.close();
      }
    };

    loadUserData();
  }, [sessionToken]);

  const goToPage = (page: string) => {
    router.push(page);
  };

  const getLogoDimensions = () => {
    if (Platform.OS === 'web') {
      return {
        width: Math.min(screenWidth * 0.4, 500),
        height: Math.min(screenWidth * 0.12, 150),
      };
    } else {
      return {
        width: Math.min(screenWidth * 0.7, 300),
        height: Math.min(screenWidth * 0.21, 90),
      };
    }
  };

  const logoDimensions = getLogoDimensions();

  return (
    <View style={[styles.homeContainer, { backgroundColor }]}>
      <View style={[
        styles.logoContainer,
        Platform.OS === 'web' ? styles.logoContainerWeb : styles.logoContainerMobile
      ]}>
        <Image
          source={require('../assets/images/GreenPinkLogo.png')}
          style={[styles.logo, { width: logoDimensions.width, height: logoDimensions.height }]}
          resizeMode="contain"
        />
      </View>

      <View style={[styles.avatar, { marginTop: 100 }]}>
        <Image
          source={avatar_list.find(avatar => avatar.avatar_id === avatarImage)?.avatar_image}
          style={{ width: 250, height: 250 }}
          resizeMode="contain"
        />
      </View>

      <View style={styles.buttonContainer}>
        <TouchableOpacity
          style={[styles.buttonStyle, { backgroundColor: getDarkerShade(backgroundColor) }]}
          onPress={() => goToPage('./journalPage/journalPage')}
        >
          <Text style={styles.buttonText}>Journal</Text>
          <Svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            width="24"
            height="24"
            strokeWidth="2"
            style={styles.icon}
          >
            <Path d="M6 4h11a2 2 0 0 1 2 2v12a2 2 0 0 1 -2 2h-11a1 1 0 0 1 -1 -1v-14a1 1 0 0 1 1 -1m3 0v18" />
            <Path d="M13 8l2 0" />
            <Path d="M13 12l2 0" />
          </Svg>
        </TouchableOpacity>

        {/* Profile Button */}
        <TouchableOpacity
          style={[styles.buttonStyle, { backgroundColor: getDarkerShade(backgroundColor) }]}
          onPress={() => goToPage('./profilePage/profilePage')}
        >
          <Text style={styles.buttonText}>Profile</Text>
          <Svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            width="24"
            height="24"
            strokeWidth="2"
            style={styles.icon}
          >
            <Path d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4z" />
            <Path d="M4 19c0-2.21 3.58-4 8-4s8 1.79 8 4" />
          </Svg>
        </TouchableOpacity>

        {/* Store Button */}
        <TouchableOpacity
          style={[styles.buttonStyle, { backgroundColor: getDarkerShade(backgroundColor) }]}
          onPress={() => goToPage('./storePage/storePage')}
        >
          <Text style={styles.buttonText}>Store</Text>
          <Svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            width="24"
            height="24"
            strokeWidth="2"
            style={styles.icon}
          >
            <Path d="M6 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0" />
            <Path d="M17 19m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0" />
            <Path d="M17 17h-11v-14h-2" />
            <Path d="M6 5l14 1l-1 7h-13" />
          </Svg>
        </TouchableOpacity>

        {/* Closet Button */}
        <TouchableOpacity
          style={[styles.buttonStyle, { backgroundColor: getDarkerShade(backgroundColor) }]}
          onPress={() => goToPage('./closetPage/closetPage')}
        >
          <Text style={styles.buttonText}>Closet</Text>
          <Svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 24 24"
            fill="none"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
            width="24"
            height="24"
            strokeWidth="2"
            style={styles.icon}
          >
            <Path d="M14 6a2 2 0 1 0 -4 0c0 1.667 .67 3 2 4h-.008l7.971 4.428a2 2 0 0 1 1.029 1.749v.823a2 2 0 0 1 -2 2h-14a2 2 0 0 1 -2 -2v-.823a2 2 0 0 1 1.029 -1.749l7.971 -4.428" />
          </Svg>
        </TouchableOpacity>

        {/* Chat Button */}
        {enableChat && (
          <TouchableOpacity style={[styles.buttonStyle, { backgroundColor: getDarkerShade(backgroundColor) }]}
            onPress={() => goToPage('./chatPage/chatPage')}
          >
            <Text style={styles.buttonText}>Chat</Text>
            <Svg
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 24 24"
              fill="none"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              width="24"
              height="24"
              strokeWidth="2"
              style={styles.icon}
            >
              <Path d="M21 15a2 2 0 0 1 -2 2h-4l-4 4v-4h-4a2 2 0 0 1 -2 -2v-10a2 2 0 0 1 2 -2h12a2 2 0 0 1 2 2z" />
            </Svg>
          </TouchableOpacity>
        )}

        {/* Friends Button */}
        {allowAddViewFriends && (
          <TouchableOpacity style={[styles.buttonStyle, { backgroundColor: getDarkerShade(backgroundColor) }]}
            onPress={() => goToPage('./friendsPage/friendsPage')}
          >
            <Text style={styles.buttonText}>Friends</Text>
            <Svg
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 24 24"
              fill="none"
              stroke="currentColor"
              strokeLinecap="round"
              strokeLinejoin="round"
              width="24"
              height="24"
              strokeWidth="2"
              style={styles.icon}
            >
              <Path d="M7 5m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0" />
              <Path d="M5 22v-5l-1 -1v-4a1 1 0 0 1 1 -1h4a1 1 0 0 1 1 1v4l-1 1v5" />
              <Path d="M17 5m-2 0a2 2 0 1 0 4 0a2 2 0 1 0 -4 0" />
              <Path d="M15 22v-4h-2l2 -6a1 1 0 0 1 1 -1h2a1 1 0 0 1 1 1l2 6h-2v4" />
            </Svg>
          </TouchableOpacity>
        )}
      </View>
    </View>
  );
};

export default HomePage;