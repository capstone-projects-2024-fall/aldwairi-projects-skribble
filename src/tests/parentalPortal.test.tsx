import React from 'react';
import { render, fireEvent, waitFor } from '@testing-library/react-native';
import ParentalPortal from '../app/parentalPortal/parentalPortal';
import { AuthContext } from '../app/AuthContext';
import createNeo4jDriver from '../app/utils/databaseSetUp';
import { Alert } from 'react-native';

jest.mock('../app/utils/databaseSetUp', () => ({
    __esModule: true,
    default: jest.fn().mockReturnValue({
        session: jest.fn().mockReturnValue({
            run: jest.fn().mockResolvedValue({}),
            close: jest.fn(),
        }),
    }),
}));

jest.spyOn(Alert, 'alert');

const mockAuthContext = {
    sessionToken: 'mockSessionToken',
};

const wrapper = ({ children }: { children: React.ReactNode }) => (
    <AuthContext.Provider value={mockAuthContext}>
        {children}
    </AuthContext.Provider>
);

describe('ParentalPortal Component', () => {
    // Test 1: Renders the parental portal page correctly
    it('renders the parental portal page correctly', () => {
        const { getByText } = render(<ParentalPortal />, { wrapper });

        // Check if key elements exist in the component
        expect(getByText('Change Email')).toBeTruthy();
        expect(getByText('Allow Add/View Friends')).toBeTruthy();
        expect(getByText('Enable Chat Feature')).toBeTruthy();
        expect(getByText('Set Time Limit (hours per day):')).toBeTruthy();
    });

    it('matches the snapshot', () => {
        const { toJSON } = render(<ParentalPortal />, { wrapper });

        // Generate and match snapshot
        expect(toJSON()).toMatchSnapshot();
    });

    // Test 2: Buttons and toggles are interactable
    it('confirms buttons and toggles are interactable', () => {
        const { getByText, getByLabelText } = render(<ParentalPortal />, { wrapper });

        // Simulate pressing the "Change Email" button
        fireEvent.press(getByText('Change Email'));

        // Simulate toggling a switch
        const friendsToggle = getByLabelText('Allow Add/View Friends');
        fireEvent(friendsToggle, 'valueChange', true); // For toggle switches, use 'valueChange'
    });


    // Test 3: Changing parent email
    it('allows changing the parent email', async () => {
        const { getByPlaceholderText, getByText } = render(<ParentalPortal />, { wrapper });

        const emailInput = getByPlaceholderText('Enter new email');
        const changeEmailButton = getByText('Change Email');

        // Simulate entering a new email
        fireEvent.changeText(emailInput, 'newparentemail@example.com');

        // Simulate pressing the button
        fireEvent.press(changeEmailButton);

        await waitFor(() => {
            expect(Alert.alert).toHaveBeenCalledWith('Success', 'Email updated successfully');
        });
    });


    // Test 4: Toggle "Allow Add/View Friends"
    it('toggles "Allow Add/View Friends"', async () => {
        const { getByLabelText } = render(<ParentalPortal />, { wrapper });

        const friendsToggle = getByLabelText('Allow Add/View Friends');

        // Simulate toggling the switch on
        fireEvent(friendsToggle, 'valueChange', true);

        await waitFor(() => {
            expect(Alert.alert).toHaveBeenCalledWith('Settings updated', 'Friends view toggled.');
        });
    });

    // Test 5: Enable "Chat Feature"
    it('toggles "Enable Chat Feature"', async () => {
        const { getByLabelText } = render(<ParentalPortal />, { wrapper });

        const chatToggle = getByLabelText('Enable Chat Feature');

        // Simulate toggling the switch on
        fireEvent(chatToggle, 'valueChange', true);

        await waitFor(() => {
          expect(Alert.alert).toHaveBeenCalledWith('Settings updated', 'Chat feature toggled.');
        });
      });

    // Test 6: Toggle "Allow Media Sharing"
    it('toggles "Allow Media Sharing"', async () => {
        const { getByLabelText } = render(<ParentalPortal />, { wrapper });

        const mediaSharingToggle = getByLabelText('Allow Media Sharing');

        // Simulate toggling the switch on
        fireEvent(mediaSharingToggle, 'valueChange', true);

        await waitFor(() => {
          expect(Alert.alert).toHaveBeenCalledWith('Settings updated', 'Media Sharing toggled.');
        });
      });


    // Test 7: Change time limit in the field
    it('allows changing the time limit', async () => {
        const { getByLabelText, getByText } = render(<ParentalPortal />, { wrapper });

        const timeLimitField = getByLabelText('Set Time Limit (hours per day):');
        const saveButton = getByText('Save Time Limit');

        fireEvent.changeText(timeLimitField, '2');
        fireEvent.press(saveButton);

        await waitFor(() => {
            expect(Alert.alert).toHaveBeenCalledWith('Settings updated', 'Time limit set to 2 hours.');
        });
    });
});
