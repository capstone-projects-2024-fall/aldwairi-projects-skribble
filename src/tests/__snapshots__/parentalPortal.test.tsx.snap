// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParentalControlPanel Component matches the snapshot 1`] = `
<RCTScrollView
  style={
    [
      {
        "flex": 1,
        "padding": 70,
        "paddingRight": 80,
      },
      {
        "backgroundColor": "#FFFFFF",
      },
    ]
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "height": 200.1,
          "justifyContent": "center",
          "marginBottom": 10,
          "width": "100%",
        }
      }
    >
      <Image
        resizeMode="contain"
        source={1}
        style={
          {
            "height": "100%",
            "maxHeight": 100,
            "width": 600,
          }
        }
      />
    </View>
    <Text
      style={
        {
          "color": "#333",
          "fontSize": 36,
          "fontWeight": "bold",
          "marginTop": -30,
          "marginVertical": 20,
          "textAlign": "center",
        }
      }
    >
      Parental Controls
    </Text>
    <View
      style={
        {
          "marginBottom": 30,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
            "marginBottom": 10,
          }
        }
      >
        <Text
          style={
            {
              "fontWeight": "bold",
            }
          }
        >
          Current Email: 
        </Text>
      </Text>
      <Text
        style={
          {
            "fontSize": 18,
            "fontWeight": "500",
            "marginBottom": 5,
          }
        }
      >
        Change Email:
      </Text>
      <TextInput
        autoCapitalize="none"
        keyboardType="email-address"
        onChangeText={[Function]}
        placeholder="Enter new email"
        style={
          {
            "backgroundColor": "#fff",
            "borderColor": "#ddd",
            "borderRadius": 8,
            "borderWidth": 1,
            "fontSize": 16,
            "marginBottom": 10,
            "padding": 10,
          }
        }
        value=""
      />
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ebebeb",
            "borderRadius": 8,
            "borderWidth": 3,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 200,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Change Email
        </Text>
      </View>
    </View>
    <View
      style={
        {
          "marginBottom": 30,
        }
      }
    >
      <Text
        style={
          {
            "color": "#333",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 15,
          }
        }
      >
        Feature Controls
      </Text>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "500",
              "marginBottom": 5,
            }
          }
        >
          Allow Add/View Friends
        </Text>
        <RCTSwitch
          accessibilityLabel="Allow Add/View Friends"
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#81b0ff"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#f4f3f4"
          tintColor="#767577"
          value={false}
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "500",
              "marginBottom": 5,
            }
          }
        >
          Enable Chat Feature
        </Text>
        <RCTSwitch
          accessibilityLabel="Enable Chat Feature"
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#81b0ff"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#f4f3f4"
          tintColor="#767577"
          value={false}
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "500",
              "marginBottom": 5,
            }
          }
        >
          Allow Media Sharing
        </Text>
        <RCTSwitch
          accessibilityLabel="Allow Media Sharing"
          accessibilityRole="switch"
          onChange={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          onTintColor="#81b0ff"
          style={
            {
              "height": 31,
              "width": 51,
            }
          }
          thumbTintColor="#f4f3f4"
          tintColor="#767577"
          value={false}
        />
      </View>
      <View
        style={
          {
            "alignItems": "center",
            "flexDirection": "row",
            "justifyContent": "space-between",
            "marginBottom": 15,
            "paddingVertical": 5,
          }
        }
      >
        <Text
          style={
            {
              "fontSize": 18,
              "fontWeight": "500",
              "marginBottom": 5,
            }
          }
        >
          Set Time Limit (hours per day):
        </Text>
        <TextInput
          keyboardType="numeric"
          maxLength={2}
          onChangeText={[Function]}
          style={
            {
              "backgroundColor": "#fff",
              "borderColor": "#ddd",
              "borderRadius": 8,
              "borderWidth": 1,
              "fontSize": 16,
              "padding": 10,
              "textAlign": "center",
              "width": 80,
            }
          }
          value="2"
        />
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "backgroundColor": "#ebebeb",
            "borderRadius": 8,
            "borderWidth": 3,
            "marginVertical": 10,
            "opacity": 1,
            "padding": 15,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": 200,
          }
        }
      >
        <Text
          style={
            {
              "color": "#000",
              "fontSize": 18,
              "fontWeight": "600",
            }
          }
        >
          Save Settings
        </Text>
      </View>
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "#ebebeb",
          "borderRadius": 8,
          "borderWidth": 3,
          "marginVertical": 10,
          "opacity": 1,
          "padding": 15,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.25,
          "shadowRadius": 4,
          "width": 75,
        }
      }
    >
      <Text
        style={
          {
            "color": "#000",
            "fontSize": 18,
            "fontWeight": "600",
          }
        }
      >
        Back
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
